core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/PushLimitThroughOffset.java

```diff
@@ -46,6 +49,7 @@
     private static final Capture<OffsetNode> CHILD = newCapture();
 
     private static final Pattern<LimitNode> PATTERN = limit()
+            .with(isOrderSensitive().equalTo(false))
```

Comment:
Offset doesn't change the order of the input, so this should be allowed

Reply:
Making this change.
Should there be a PushTopNThroughOffset rule as well ?